### PR TITLE
Arvato Rss ECO release

### PR DESCRIPTION
**Pull request by ghost**
- Developer(s): @sergeysikachev

- Peer Reviewer:

- Ticket: URL_HERE

- Project PR: URL_HERE

- Academy PR: URL_HERE (for feature documentation and if needed migration guides and project integration guide)

#### Please confirm

- [ ] No new OS components - or they have been approved by legal department
- [ ] All new or heavily changed classes get an "A" rating (Scrutinizer), except Facades, Factories, QueryContainers and Tests

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraints         |
   :--------------------- | :------------------------ | :--------------------- |
   ArvatoRss              |       major                 |                       |

#### Release Notes

- Arvato Rss  Risk Check integration.
- Customer data, like address, order info etc. is used to check customer for fraud using Arvato RSS SOAP based web service.
-----------------------------------------

#### Bundle ArvatoRss

_**Major:** Backwards-compatible feature_

Def of done (by responsible developer):
- [ ] All changes are backward-compatible. Outdated code is marked as deprecated
- [ ] The change is isolated and not mixed with any cleanups or other changes
- [ ] New code fits to the architecture rules
- [ ] All new or changed business logic is covered by functional tests (in Zed business layer)

##### Change log

Initial Release:

- Arvato Rss  Risk Check integration.
- Customer data, like address, order info etc. is used to check customer for fraud using Arvato RSS SOAP based web service.

### CODE REVIEW

File: src/SprykerEco/Service/ArvatoRss/Iso3166Converter.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+namespace SprykerEco\Service\ArvatoRss;
+
+use Spryker\Service\Kernel\AbstractService;
+
+class Iso3166Converter extends AbstractService implements Iso3166ConverterInterface
+{
+
+    /**
+     * @const string ISO2_KEY
```
Comment (dereuromark) [inline src/SprykerEco/Service/ArvatoRss/Iso3166Converter.php]: no need to repeat the key here (ISO2_KEY)

File: config/config.dist.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+use SprykerEco\Shared\ArvatoRss\ArvatoRssConstants;
+
+$config[ArvatoRssConstants::ARVATORSS] = [
```
Comment (dereuromark) [inline config/config.dist.php]: We usually never have this nesting in core, do we need it?
The constants itself are already unique: `const ARVATO_RSS_URL = 'ARVATO_RSS:ARVATO_RSS_URL'` etc

File: src/SprykerEco/Zed/ArvatoRss/Business/Api/Adapter/SoapApiAdapter.php [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+use SprykerEco\Zed\ArvatoRss\Business\Api\Converter\RiskCheckRequestConverterInterface;
+use SprykerEco\Zed\ArvatoRss\Business\Api\Converter\RiskCheckRequestHeaderConverterInterface;
+use SprykerEco\Zed\ArvatoRss\Business\Api\Converter\RiskCheckResponseConverterInterface;
+use SprykerEco\Zed\ArvatoRss\Business\Api\Exception\ArvatoRssRiskCheckApiException;
+
+class SoapApiAdapter implements ApiAdapterInterface
+{
+
+    /**
+     * @const WSDL_PATH
```
Comment (dereuromark) [inline src/SprykerEco/Zed/ArvatoRss/Business/Api/Adapter/SoapApiAdapter.php]: I would remove this invalid doc block alltogether, for consts there is no reason to annotate without actual description content.

File: src/SprykerEco/Zed/ArvatoRss/Business/Api/Converter/RiskCheckRequestConverter.php [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+     * @return array
+     */
+    public function convert(ArvatoRssRiskCheckRequestTransfer $arvatoRssRiskCheckRequestTransfer)
+    {
+        $result = [];
+
+        $billingCustomer = $this->convertBillingCustomer($arvatoRssRiskCheckRequestTransfer);
+        $order = $this->convertOrder($arvatoRssRiskCheckRequestTransfer);
+
+        $result = array_merge($result, $billingCustomer);
```
Comment (dereuromark) [inline src/SprykerEco/Zed/ArvatoRss/Business/Api/Converter/RiskCheckRequestConverter.php]: For assoc arrays one usually uses + operator for merge. This is safer in terms of keeping the correct keys (array_merge is about the values only).

**Review by dereuromark — CHANGES_REQUESTED** [review]

File: src/SprykerEco/Service/ArvatoRss/Iso3166Converter.php (hunk 1 above)
Comment (ghost) [reply src/SprykerEco/Service/ArvatoRss/Iso3166Converter.php]: Removed comments from constants here.

File: config/config.dist.php (hunk 2 above)
Comment (ghost) [reply config/config.dist.php]: Changed configuration to one-level config.

File: src/SprykerEco/Zed/ArvatoRss/Business/Api/Adapter/SoapApiAdapter.php (hunk 3 above)
Comment (ghost) [reply src/SprykerEco/Zed/ArvatoRss/Business/Api/Adapter/SoapApiAdapter.php]: Removed.

File: src/SprykerEco/Zed/ArvatoRss/Business/Api/Converter/RiskCheckRequestConverter.php (hunk 4 above)
Comment (ghost) [reply src/SprykerEco/Zed/ArvatoRss/Business/Api/Converter/RiskCheckRequestConverter.php]: Refactored array_merge.

**Pull request merged**